config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1266,7 +1285,19 @@ public void setBearerTokenConverter(ServerAuthenticationConverter bearerTokenCon
 			 */
 			public JwtSpec authenticationManager(ReactiveAuthenticationManager authenticationManager) {
 				Assert.notNull(authenticationManager, "authenticationManager cannot be null");
-				this.authenticationManager = authenticationManager;
+				this.authenticationManagerResolver = request -> Mono.just(authenticationManager);
+				return this;
+			}
+
+			/**
+			 * Configures the {@link ReactiveAuthenticationManagerResolver} to use
+			 * @param authenticationManagerResolver the authentication manager resolver
+			 * @return the {@code JwtSpec} for additional configuration
+			 */
+			public JwtSpec authenticationManager(
```

Comment:
This method shouldn't be necessary. The user can supply an `authenticationManagerResolver` at the `oauth2` level or an `authenticationManager` at the `jwt` or `opaqueToken` levels.